File: backend-quarkus/src/main/java/app/nanotes/backend/web/QueryParams.java
package app.nanotes.backend.web;

import app.nanotes.backend.apperr.ValidationException;

final class QueryParams {
    private QueryParams() {}

    static int parsePositiveInt(String raw, String errorMessage) {
        int v = parseInt(raw, errorMessage);
        if (v <= 0) {
            throw new ValidationException(errorMessage);
        }
        return v;
    }

    static int parseInt(String raw, String errorMessage) {
        try {
            return Integer.parseInt(raw);
        } catch (NumberFormatException e) {
            throw new ValidationException(errorMessage);
        }
    }
}

File: backend-quarkus/src/main/kotlin/app/nanotes/backend/web/QueryParams.kt
package app.nanotes.backend.web

import app.nanotes.backend.apperr.ValidationException

internal object QueryParams {
    fun parsePositiveInt(raw: String, errorMessage: String): Int {
        val v = parseInt(raw, errorMessage)
        if (v <= 0) throw ValidationException(errorMessage)
        return v
    }

    fun parseInt(raw: String?, errorMessage: String): Int =
        raw?.toIntOrNull() ?: throw ValidationException(errorMessage)
}
